test/content/UniqueContentTests.js

```diff
@@ -0,0 +1,338 @@
+const { expect } = require("chai");
+const { ethers, upgrades } = require("hardhat");
+const { sign } = require("../mint");
+
+describe('Unique Content Contract Tests', () => {
+    var uniqueContent
+    var deployerAddress, deployerAltAddress, receiverAddress, playerAddress, player2Address;
+    var AccessControlManager, ContentStorage, Content;
+    var content;
+    var contentStorage;
+    var accessControlManager;
+    var asset;
+    var UniqueContent;
+
+    before(async () => {
+        [deployerAddress, deployerAltAddress, receiverAddress, playerAddress, player2Address] = await ethers.getSigners();
+        UniqueContent = await ethers.getContractFactory("UniqueContent");
+        AccessControlManager = await ethers.getContractFactory("AccessControlManager");
+        ContentStorage = await ethers.getContractFactory("ContentStorage");
+        Content = await ethers.getContractFactory("Content");
+        asset = [
+            ["arweave.net/tx/public-uri-0", "", ethers.constants.MaxUint256, deployerAddress.address, 20000],
+            ["arweave.net/tx/public-uri-1", "", 100, ethers.constants.AddressZero, 0],
+        ];
+    });
+
+    beforeEach(async () => {
+        accessControlManager = await upgrades.deployProxy(AccessControlManager, []);
+        contentStorage = await upgrades.deployProxy(ContentStorage, [deployerAltAddress.address, 12000, "arweave.net/tx-contract-uri"]);
+        content = await upgrades.deployProxy(Content, [contentStorage.address, accessControlManager.address]);
+
+        await contentStorage.grantRole(await contentStorage.DEFAULT_ADMIN_ROLE(), content.address);
+
+        // give deployer address approval; This would normally be done through the ContentManager
+        minter_role = await accessControlManager.MINTER_ROLE();
+        await accessControlManager.grantRole(minter_role, deployerAddress.address);
+
+        // Set the content contract as the new parent
+        await accessControlManager.setParent(content.address);
+        await contentStorage.addAssetBatch(asset);
+
+        // playerAddress mints 2 of asset 0 and 1 of asset 1
+        const signature = await sign(playerAddress.address, [0, 1], [2, 1], 1, deployerAddress.address, content.address);
+        var mintData = [playerAddress.address, [0, 1], [2, 1], 1, deployerAddress.address, signature];
+        await content.connect(playerAddress).mintBatch(mintData);
+
+        // launch unique content contract
+        uniqueContent = await upgrades.deployProxy(UniqueContent);
+        
+        // Give unique content contract permission to transfer original asset
+        await content.connect(playerAddress).setApprovalForAll(uniqueContent.address, true);
+    });
+
+    describe("Basic Tests", () => {
+        // Todo: Do interface checks
+    });
+
+    describe("Mint Tokens", () => {
```

Comment:
a bunch of these tests need to get updated because of the changes above. I'll take a look at this further when everything is ready.